File: sample_decoder.go
package byteSample

import (
	"github.com/bytebuffer_parser/parsers"
	"github.com/golang/snappy"
	"log"
)

func SAMPLE_Decoder(compression bool, byteArr []byte) [][]Person {

	bb := parsers.Buffer{
		FloatIntEncoderVal: 10000.0,
		Endian:             "big",
	}

	if compression {
		decompressedData, err := snappy.Decode(nil, byteArr)

		if err != nil {
			log.Fatal("Failed to decompress data...")
		}

		bb.Wrap(decompressedData)
	} else {
		bb.Wrap(byteArr)
	}

	arrLen := bb.GetShort()
	obj := make([][]Person, arrLen)

	for i0 := 0; i0 < arrLen; i0++ {

		arrLen1 := bb.GetShort()

		obj[i0] = make([]Person, arrLen1)

		for i1 := 0; i1 < arrLen1; i1++ {

			obj[i0][i1].Epoch = bb.GetLongInteger()

			WatchArrLen0 := bb.GetShort()
			obj[i0][i1].Watch = make([][]int, WatchArrLen0)
			for index00 := 0; index00 < WatchArrLen0; index00++ {

				WatchArrLen1 := bb.GetShort()
				obj[i0][i1].Watch[index00] = make([]int, WatchArrLen1)
				for index01 := 0; index01 < WatchArrLen1; index01++ {

					obj[i0][i1].Watch[index00][index01] = bb.GetInteger()

				}

			}

			obj[i0][i1].Xyz = bb.GetInteger()

			obj[i0][i1].Salary = bb.GetFloatUsingIntEncoding()

			EmployeeArrLen0 := bb.GetShort()
			obj[i0][i1].Employee = make([][][][]Employees, EmployeeArrLen0)
			for index00 := 0; index00 < EmployeeArrLen0; index00++ {

				EmployeeArrLen1 := bb.GetShort()
				obj[i0][i1].Employee[index00] = make([][][]Employees, EmployeeArrLen1)
				for index11 := 0; index11 < EmployeeArrLen1; index11++ {

					EmployeeArrLen2 := bb.GetShort()
					obj[i0][i1].Employee[index00][index11] = make([][]Employees, EmployeeArrLen2)
					for index22 := 0; index22 < EmployeeArrLen2; index22++ {

						EmployeeArrLen3 := bb.GetShort()
						obj[i0][i1].Employee[index00][index11][index22] = make([]Employees, EmployeeArrLen3)
						for index33 := 0; index33 < EmployeeArrLen3; index33++ {

							obj[i0][i1].Employee[index00][index11][index22][index33].Name = bb.GetString()

							obj[i0][i1].Employee[index00][index11][index22][index33].Salary = bb.GetFloatUsingIntEncoding()

							StudentArrLen0 := bb.GetShort()
							obj[i0][i1].Employee[index00][index11][index22][index33].Student = make([]StudentClass, StudentArrLen0)
							for index40 := 0; index40 < StudentArrLen0; index40++ {

								obj[i0][i1].Employee[index00][index11][index22][index33].Student[index40].Name = bb.GetString()

							}

						}

					}

				}

			}

		}

	}

	return obj
}

File: sample_decoder.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace byteSample {

  class Decoder {

    public SAMPLE_Decoder(byte[] byteArr)[][] Person {

      ByteBuffer bb = new ByteBuffer();

      bb.Wrap(byteArr);

      int arrLen = (int) bb.GetShort();
      Person[][] obj = new Person[arrLen][];

      for (int i0 = 0; i0 < arrLen; i0++) {

        int arrLen1 = (int) bb.GetShort();
        obj[i0] = new Person[arrLen1];
        for (int i1 = 0; i1 < arrLen1; i1++) {

          obj[i0][i1].Epoch = bb.GetLong();
          int WatchArrLen0 = (int) bb.GetShort();
          int[][] obj[i0][i1].Watch = new int[WatchArrLen0][];
          for (int index00: = 0; index00 < WatchArrLen0; index00++) {

            int WatchArrLen1: = (int) bb.GetShort();
            int[] obj[i0][i1].Watch[index00] = new int[WatchArrLen1];
            for (int index01: = 0; index01 < WatchArrLen1; index01++) {

              obj[i0][i1].Watch[index00][index01] = bb.GetInt();

            }

          }

          obj[i0][i1].Xyz = bb.GetInt();

          obj[i0][i1].Salary = bb.GetDouble();

          int EmployeeArrLen0 = (int) bb.GetShort();
          Employees obj[i0][i1].Employee = new Employees[EmployeeArrLen0][][][];
          for (int index00 = 0; index00 < EmployeeArrLen0; index00++) {

            int EmployeeArrLen1 = (int) bb.GetShort();
            Employees obj[i0][i1].Employee[index00] = new Employees[EmployeeArrLen1][][];
            for (int index11 = 0; index11 < EmployeeArrLen1; index11++) {

              int EmployeeArrLen2 = (int) bb.GetShort();
              Employees obj[i0][i1].Employee[index00][index11] = new Employees[EmployeeArrLen2][];
              for (int index22 = 0; index22 < EmployeeArrLen2; index22++) {

                int EmployeeArrLen3 = (int) bb.GetShort();
                Employees obj[i0][i1].Employee[index00][index11][index22] = new Employees[EmployeeArrLen3];
                for (int index33 = 0; index33 < EmployeeArrLen3; index33++) {

                  obj[i0][i1].Employee[index00][index11][index22][index33].Name = bb.GetString();

                  obj[i0][i1].Employee[index00][index11][index22][index33].Salary = bb.GetDouble();

                  int StudentArrLen0 = (int) bb.GetShort();
                  StudentClass obj[i0][i1].Employee[index00][index11][index22][index33].Student = new StudentClass[StudentArrLen0];
                  for (int index40 = 0; index40 < StudentArrLen0; index40++) {

                    obj[i0][i1].Employee[index00][index11][index22][index33].Student[index40].Name = bb.GetString();

                  }

                }

              }

            }

          }

        }

      }

      bb.Dispose();

      return obj;
    }
  }
}